Sources/MlemMiddleware/FeedLoaders/User Content Feed Loader/UserContentFeedLoader.swift

```diff
@@ -0,0 +1,240 @@
+    var comments: [Comment2] { tempCommentStream?.items ?? commentStream.items }
+    
+    public init(
+        api: ApiClient,
+        userId: Int,
+        sortType: FeedLoaderSort.SortType,
+        savedOnly: Bool,
+        smallAvatarSize: CGFloat,
+        largeAvatarSize: CGFloat,
+        urlCache: URLCache
+    ) {
+        self.api = api
+        self.userId = userId
+        self.sortType = sortType
+        self.savedOnly = savedOnly
+        self.apiPage = 0
+        self.contentPage = 0
+        self.items = .init()
+        self.contentLoadingSemaphore = .init(value: 1)
+        self.loadingState = .idle
+        self.postStream = .init()
+        self.commentStream = .init()
+        self.smallAvatarIconSize = Int(smallAvatarSize * 2)
+        self.largeAvatarIconSize = Int(largeAvatarSize * 2)
+        self.urlCache = urlCache
+    }
+    
+    // MARK: Public Methods
+    
+    // protocol conformance
+    public func loadIfThreshold(_ item: UserContent) throws {
+        try loadIfThreshold(item, asChild: false)
+    }
+    
+    /// Given a UserContent, loads more items if that content is a threshold item
+    /// - Parameters:
+    ///   - item: UserContent
+    ///   - loadChildOnly: if true, the item will be evaluated against the relevant stream threshold rather than the parent threshold
+    public func loadIfThreshold(_ item: UserContent, asChild: Bool) throws {
+        var shouldLoad: Bool = false
+        if asChild {
+            shouldLoad = switch item.wrappedValue {
+            case let .post(post): postStream.thresholds.isThreshold(post)
+            case let .comment(comment): commentStream.thresholds.isThreshold(comment)
+            }
+        } else {
+            shouldLoad = thresholds.isThreshold(item)
+        }
+        
+        // regardless of which threshold triggers this, always call loadMoreItems() because there's no item-specific endpoint
+        if shouldLoad {
+            Task(priority: .userInitiated) {
+                try await loadMoreItems()
+            }
+        }
+    }
+    
+    public func loadMoreItems() async throws {
+        print("Loading more user content")
+        try await loadContentPage(contentPage + 1)
+    }
+    
+    public func changeSortType(to sortType: FeedLoaderSort.SortType) {
+        self.sortType = sortType
+        items = .init()
+        postStream = .init()
+        commentStream = .init()
+    }
+    
+    public func refresh(clearBeforeRefresh: Bool) async throws {
+        loadingState = .loading
+        
+        if clearBeforeRefresh {
+            items = .init()
+        } else {
+            tempPostStream = postStream
+            tempCommentStream = commentStream
+        }
+        postStream = .init()
+        commentStream = .init()
+        apiPage = 0
+        contentPage = 0
+        
+        try await loadMoreItems()
+        
+        tempPostStream = nil
+        tempCommentStream = nil
```

Comment:
If an error is thrown here the temp streams won't be erased

```suggestion
        defer {
            tempPostStream = nil
            tempCommentStream = nil
        }
        try await loadMoreItems()
```

Reply:
Nice catch!